drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -441,6 +441,24 @@ def assessment_date_item
     @assessment_date_item ||= link_id_item_hash.values.find(&:assessment_date)
   end
 
+  # Currency is an integer or float
+  CURRENCY_RGX = /\A-?[1-9]\d*\z|\A-?[1-9]\d*\.\d+\z|\A-?0\.\d+\z/
+  INTEGER_RGX = /\A-?[1-9]\d*\z|\A0\z/
+
+  def validate_input_format(item, value)
+    return unless value.present?
+    # special case values?
+    return if value == 'DATA_NOT_COLLECTED'
+    return if value == '_HIDDEN'
+
+    case item.type
+    when 'INTEGER'
+      ['not a valid integer'] unless value.to_s.strip =~ INTEGER_RGX
```

Comment:
perhaps this is too far? the bug in question only deals with currency fields but it seems like we shouldn't except non-numeric values for integers either